File: lib/js/Architecture_Lib/MCFactory.js
console.log("MCFactory.js loaded");

var MCFactory = function() {};

MCFactory.createSimpleMc = function(name) {
	var o = {};
	return o;
}

MCFactory.createComplexMc = function(name) {
	var o = {};
	o.controller = new Controller(name);
	return o;
}

File: lib/java/architecture/MCFactory.java
package architecture;

public class MCFactory {
	public static Object createSimpleMc(String name) {
		// Not implemented yet
		return null;
	}
	
	public static ComplexMc createComplexMc(String componentName) {
		// Not implemented yet
		ComplexMc complexMc = new ComplexMc(componentName);
		
		Controller controller = new Controller(componentName);
		complexMc.setController(controller);
		return complexMc;
	}
}
